eventuals/task.h

```diff
@@ -544,15 +610,9 @@ struct Task {
     };
   }
 
-  template <typename Error>
+  template <typename To, typename Error>
   static auto Failure(Error error) {
-    // TODO(benh): optimize away heap allocation.
-    return [error = std::make_unique<Error>(std::move(error))]() mutable {
-      return Eventual<_TaskFailure>()
-          .start([&](auto& k) mutable {
-            k.Fail(Error(std::move(*error)));
-          });
-    };
+    return TaskFromToWith<void, To>(std::make_exception_ptr(std::move(error)));
```

Comment:
Let's just revert this since then we don't have to specify `To` and both implementations do dynamic memory allocation.